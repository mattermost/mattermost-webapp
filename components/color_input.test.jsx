// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import ColorInput from 'components/color_input.jsx';

describe('components/ColorInput', () => {
    test('should match snapshot, init', () => {
        const wrapper = shallow(
            <ColorInput color='#ffffff'/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, opened', () => {
        const wrapper = shallow(
            <ColorInput color='#ffffff'/>
        );

        wrapper.find('.input-group-addon').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, toggle picker', () => {
        const wrapper = shallow(
            <ColorInput color='#ffffff'/>
        );

        wrapper.find('.input-group-addon').simulate('click');
        wrapper.find('.input-group-addon').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, click on picker', () => {
        const wrapper = shallow(
            <ColorInput color='#ffffff'/>
        );

        wrapper.find('.input-group-addon').simulate('click');
        wrapper.find('.color-popover').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    test('should have match state on togglePicker', () => {
        const wrapper = shallow(
            <ColorInput color='#ffffff'/>
        );

        wrapper.setState({isOpened: true});

        wrapper.find('.input-group-addon').simulate('click');
        expect(wrapper.state('isOpened')).toEqual(false);

        wrapper.find('.input-group-addon').simulate('click');
        expect(wrapper.state('isOpened')).toEqual(true);
    });

    test('should have called onChange prop', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <ColorInput
                color='#ffffff'
                onChange={onChange}
            />
        );

        const newColorData = {hex: '#ccc'};
        wrapper.instance().handleChange(newColorData);

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith(newColorData.hex);
    });
});
