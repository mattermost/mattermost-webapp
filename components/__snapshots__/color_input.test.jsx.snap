// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ColorInput should match snapshot, click on picker 1`] = `
<div
  className="color-input input-group"
>
  <input
    className="form-control"
    readOnly={true}
    type="text"
    value="#ffffff"
  />
  <span
    className="input-group-addon"
    onClick={[Function]}
  >
    <i
      className="color-icon"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
  </span>
  <div
    className="color-popover"
  >
    <ColorPicker
      color="#ffffff"
      disableAlpha={false}
      onChange={[Function]}
    />
  </div>
</div>
`;

exports[`components/ColorInput should match snapshot, init 1`] = `
<div
  className="color-input input-group"
>
  <input
    className="form-control"
    readOnly={true}
    type="text"
    value="#ffffff"
  />
  <span
    className="input-group-addon"
    onClick={[Function]}
  >
    <i
      className="color-icon"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
  </span>
</div>
`;

exports[`components/ColorInput should match snapshot, opened 1`] = `
<div
  className="color-input input-group"
>
  <input
    className="form-control"
    readOnly={true}
    type="text"
    value="#ffffff"
  />
  <span
    className="input-group-addon"
    onClick={[Function]}
  >
    <i
      className="color-icon"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
  </span>
  <div
    className="color-popover"
  >
    <ColorPicker
      color="#ffffff"
      disableAlpha={false}
      onChange={[Function]}
    />
  </div>
</div>
`;

exports[`components/ColorInput should match snapshot, toggle picker 1`] = `
<div
  className="color-input input-group"
>
  <input
    className="form-control"
    readOnly={true}
    type="text"
    value="#ffffff"
  />
  <span
    className="input-group-addon"
    onClick={[Function]}
  >
    <i
      className="color-icon"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
  </span>
</div>
`;
